lib/puma/binder.rb

```diff
@@ -407,24 +412,24 @@ def inherit_unix_listener(path, fd)
     end
 
     def close_listeners
-      listeners.each do |l, io|
-        io.close unless io.closed? # Ruby 2.2 issue
-        uri = URI.parse(l)
+      @listeners.each do |l, io|
+        io.close unless io.closed?
+        uri = URI.parse l
         next unless uri.scheme == 'unix'
         unix_path = "#{uri.host}#{uri.path}"
-        File.unlink unix_path if unix_paths.include? unix_path
+        File.unlink unix_path if @unix_paths.include?(unix_path) && File.exist?(unix_path)
       end
     end
 
     def redirects_for_restart
-      redirects = listeners.map { |a| [a[1].to_i, a[1].to_i] }.to_h
+      redirects = @listeners.map { |a| [a[1].to_i, a[1].to_i] }.to_h
       redirects[:close_others] = true
       redirects
     end
 
     # @version 5.0.0
     def redirects_for_restart_env
-      listeners.each_with_object({}).with_index do |(listen, memo), i|
+      @listeners.each_with_object({}).with_index do |(listen, memo), i|
```

Comment:
Why do we have to work around the accessor method?

Reply:
This is something that is somewhat personal preference, as I doubt there's any performance difference in the general case.

First, `@listeners` is initialized.  Secondly, I haven't looked at it for a while (Puma 6 is not around the corner), but there are a lot of public methods/accessors that could be removed.  Changing our code to allow that is helpful.

### General

1. Removing accessors may break tests, but conversely, accessors added just to make testing simple may not be a good idea.  Tests can use `#instance_variable_get` if needed.
2. Reducing the public API often shows where encapsulation has been broken, often where an accessor should be replaced  with one or more 'action' methods.
3. I'll never forget breaking Capybara.
4. I'm not sure if I changed all the code in Binder to use `@listeners`.